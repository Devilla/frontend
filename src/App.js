import React from 'react';
import { browserHistory } from 'react-router';
import { AppContainer } from 'react-hot-loader';
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';
import rootSaga from './sagas';
import reducer from './ducks';
import thunk from 'redux-thunk';
import { syncHistoryWithStore } from 'react-router-redux';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import createSagaMiddleware from 'redux-saga';
import { StripeProvider } from 'react-stripe-elements';

const sagaMiddleware = createSagaMiddleware();

export const store = createStore(reducer, composeWithDevTools(
  applyMiddleware(sagaMiddleware, thunk)
));

const routerHistory = syncHistoryWithStore(browserHistory, store, {selectLocationState: state => state.get('router')});


sagaMiddleware.run(rootSaga);

const App = ({Component}) =>
  <AppContainer errorReporter={Error}>
    <Provider store={store}>
      <StripeProvider apiKey={process.env.NODE_ENV === 'production' ? process.env.REACT_APP_STRIPE_KEY : process.env.REACT_APP_DEVELOPMENT_STRIPE_KEY}>
        <Component routerHistory={routerHistory} />
      </StripeProvider>
    </Provider>
  </AppContainer>;

export default App;
