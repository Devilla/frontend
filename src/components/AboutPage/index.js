import React, { Component } from 'react';
import {Animated} from "react-animated-css";
export default class About extends Component {
     componentDidMount(){
     window.scrollTo(0, 0)
   } 
  render() {
    return (
        <div>
            <div className="page-header"><div className="page-header-overlay">
            <div className="centered page-header-container w-container">
            <Animated className="page-header-title" animationIn="fadeInDown" animationOut="fadeOut" isVisible={true}>About us</Animated>
            </div></div></div>
            <div className="section innerpage">
                <div className="container w-container">
                    <p>We started Influence with one mission: to cut down the advertising cost and to convert more customers on your website.</p>
            <p>We use your existing customers’ data on your website and show it to your visitors so that they are socially influenced by buying behavior of people on the website.</p>
            <p>Using this tool we’ve been able to help a lot of businesses in increasing their money making and get more conversions.</p>
            <p>Social influence is the next big thing in marketing for influencing buyers to buy more.</p>
            <p>Let’s start with your free to start trial and get making more money for you.
            </p>
                   
                </div>
              </div>
        </div>
    );
  }
}
