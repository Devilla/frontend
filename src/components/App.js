import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import  '../assets/stylesheets/slick.css'
import  '../assets/stylesheets/slick-theme.css'
import  '../assets/stylesheets/style.css'

import Header from './_common/Header';
import Footer from './_common/Footer';
import { Spinner } from './index.js';

class App extends Component {
  render(){
    return (
        <div className="app-container">
          <Header />

          <Spinner loading={this.props.loading} />
          <div className="content">
            {this.props.children}
          </div>
          <Footer />
        </div>
    );
  }
};

App.propTypes = {
  children: PropTypes.node
};

const mapStateToProps = state => ({
  loading: state.getIn(['loading', 'state'])
});

export default connect(mapStateToProps)(App);
