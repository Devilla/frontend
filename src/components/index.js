export { default as WebsiteHeader } from "./WebsiteHeader/WebsiteHeader";
export { default as WebsiteFooter } from "./WebsiteFooter/WebsiteFooter";
export { default as WebsiteHome } from "./WebsiteHome/WebsiteHome";
export { default as WebsiteHowItWorks } from "./WebsiteHowItWorks/WebsiteHowItWorks";
export { default as WebsiteIntegrations } from "./WebsiteIntegrations/WebsiteIntegrations";
export { default as WebsitePricing } from "./WebsitePricing/WebsitePricing";
export { default as WebsiteAbout } from "./WebsiteAbout/WebsiteAbout";
export { default as WebsiteSignIn } from "./WebsiteSignIn/WebsiteSignIn";
export { default as WebsiteSignUp } from "./WebsiteSignUp/WebsiteSignUp";
export { default as WebsiteContact } from "./WebsiteContact/WebsiteContact";
export { default as WebsiteTerms } from "./WebsiteTerms/WebsiteTerms";
export { default as WebsitePolicy } from "./WebsitePolicy/WebsitePolicy";
export { default as WebsiteDemoPage}  from "./WebsiteDemoPage/WebsiteDemoPage";
export { default as WebsiteCustStory}  from "./WebsiteCustStory/WebsiteCustStory";
export { default as WebsiteFeature}  from "./WebsiteFeature/WebsiteFeature";
export { default as Affiliate}  from "./Affiliate/Affiliate";
export { default as AffiliateRegister}  from "./Affiliate/AffiliateRegister";

export { default as Header } from "./Header/Header";
export { default as ForgetPassword } from "./ForgetPassword";
export { default as PricePage } from "./PricePage";
export { default as Common } from "./Common/function";
export { default as ResetPassword } from "./ResetPassword";
export { default as Sidebar } from "./Sidebar/Sidebar";
export { default as StatsCard } from "./StatsCard/StatsCard";
export { default as Dashboard } from './Dashboard/Dashboard';
export { default as Notification } from './Notification';
export { default as Analytics } from './Analytics/analytics';
export { default as AnalyticsProfile } from './Analytics/AnalyticsProfile';
export { default as New } from './New/new';
export { default as LoginFlow } from './LoginFlow/LoginFlow';
export { default as Spinner } from './Spinner/Spinner';
export { default as Error } from './Error/Error';
export { default as ComingSoon } from './ComingSoon/ComingSoon';
export { default as Button } from './Button/Button';
export { default as Profile } from './Profile/Profile';
export { default as BillingDetails } from './Billing/BillingDetails';
export { default as UpgradeCard } from './UpgradeCard/UpgradeCard';
export { default as UpgradePlan } from './UpgradePlan/UpgradePlan';
export { default as Integration } from './Integration/Integration';
export { default as Template } from './Template';
export { default as VerificationPage } from './VerificationPage/VerificationPage';
