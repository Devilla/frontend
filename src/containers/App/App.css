@import url('https://fonts.googleapis.com/css?family=Open+Sans:200,300,400,400i,500,600,700');
@import url('https://fonts.googleapis.com/css?family=Source+Sans+Pro:200,300,400,400i,600,700');

body {
  font-family: "Raleway, Source Sans Pro, Verdana, Helvetica", sans-serif;
}
