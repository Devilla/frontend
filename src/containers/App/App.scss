@import url('https://fonts.googleapis.com/css?family=Open+Sans:200,300,400,400i,500,600,700');
@import url('https://fonts.googleapis.com/css?family=Source+Sans+Pro:200,300,400,400i,600,700');

body {
  font-family: "Raleway, Source Sans Pro, Verdana, Helvetica", sans-serif;
  display: block;
  margin: 0px;
}

.website-app {
  .alert-warning {
    background-color: #f2dede;
    border-color: #ebccd1;
    color: #a94442;
  }
  .content {
    padding-top: 100px;
  }
  .toast-style {
    padding: 17px 7px;
    background-color: rgba(0, 0, 0, 0.9) !important;
    color: #fff !important;
    text-align: center !important;
    height: 10px !important;
    border-radius: 5px !important;
    button {
      height: 1.5em;
      color: #fff;
    }
  }

  .transition-wrapper {
    position: relative;
    z-index: 1;
    .transition-item {
      position: fixed;
      top: 0;
      right: 0;
      left: 0;
    }
  }
  .signin-container,
  .authpage,
  .websitepolicy-container,
  .websitehowitworks-container,
  .websitehome-container,
  .WebsiteTerms-container,
  .websitepricing-container,
  .website-payment-container,
  .websiteintegrations-container,
  .demopage-container,
  .website-cust-story,
  .websitecontact-container,
  .websiteabout-container,
  .WebsiteFeature-container,
  .affiliateregister-container
   {
    overflow: hidden;
    box-sizing: border-box;
    padding: 20px;
    min-height: 100vh;
    transition: 500ms ease-out;

    &.transition-leave {
      opacity: 1;
    }

    &.transition-leave.transition-leave-active {
      opacity: 1;
      transform: translate3d(100%, 0, 0);
    }
  }
}
