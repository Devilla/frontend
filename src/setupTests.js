import 'jest-enzyme';

window.matchMedia = window.matchMedia || function() {
  return {
    matches : false,
    addListener : function() {},
    removeListener: function() {}
  };
};
