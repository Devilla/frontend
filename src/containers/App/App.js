import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { ToastContainer } from 'react-toastify';
import { browserHistory } from 'react-router';
import { connect } from 'react-redux';
import { Spinner } from '../../components/index.js';
import { WebsiteHeader, WebsiteFooter } from 'components';

import { checkTokenExists } from 'ducks/auth';

import './scss/stack-interface.scss';
import './scss/socicon.css';
import './scss/iconsmind.scss';
import './scss/bootstrap.scss';
import './scss/theme.scss';
import './scss/custom.scss';
import './scss/font-raleway.scss';
import './scss/flickity.scss';
import './scss/font-sourcesanspro.scss';
import './App.scss';

class App extends Component {
  componentWillMount() {
    this.checkLogin((err) => {
      if(!err) {
        browserHistory.push('/dashboard');
      }
    });
  }

  checkLogin(callback) {
    const cookie = localStorage.getItem('authToken');
    const authToken = cookie
      ? JSON.parse(cookie)
      : null;
    if (authToken) {
      this.props.checkTokenExists(authToken);
      callback()
    } else {
      callback("not logged in")
    }
  }

  render(){
    return (
      <div className="website-app">
        <div className="basic-gradient-light" data-smooth-scroll-offset="77">
           <WebsiteHeader />
           <Spinner loading={this.props.loading} />
           <div className="content">
             {this.props.children}
           </div>
           <WebsiteFooter />
        </div>
        <ToastContainer hideProgressBar={true} />
      </div>
    );
  }

}

const mapDispatchToProps = {
  checkTokenExists
};

export default connect(null, mapDispatchToProps)(App);
