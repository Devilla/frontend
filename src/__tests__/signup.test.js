import React from 'react';
import '../setupTests';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

import { Provider } from 'react-redux';

import { WebsiteSignUp } from 'components';
import { store } from 'App';



// Snapshot for SignUp React Component
describe('>>>SignUp --- Snapshot',()=>{
  it('+++capturing Snapshot of SignUp', () => {
    const renderedValue =  renderer.create(<WebsiteSignUp store={store} />).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});

describe('check for in', ()=> {
  let wrapper;
  beforeEach ( () => {
       wrapper = shallow(<WebsiteSignUp  store={store}/>);
  })
  
  it('+++ render the DUMB component', () => {
    expect(wrapper.length).toEqual(1);
  });
  
  
  it('+++ contains website - error', () => {
    let wrapper = shallow(<WebsiteSignUp  store={store}/>);
    expect(wrapper.find('.website-error').length).toBe(4);
  });

  //This condition should get pass to the state 
    it('should respond to change event and change the state of the WebsiteSignUp Component', () => {
     wrapper.find('#email').simulate('change', {target: {name: 'email', value: 'blah@gmail.com'}});
     expect(wrapper.state('email')).toEqual('blah@gmail.com');
    })
  
    it('should respond to changed  event and change the state of the SignUp Component', () => {
     wrapper.find('#password').simulate('change', {target: {name: 'password', value: 'cats'}});
     expect(wrapper.state('password')).toEqual('cats');
    })


    //This condition should not get pass to the state.
    it('should respond to change event and change the state of the WebsiteSignUp Component', () => {
      wrapper.find('#email').simulate('change', {target: {name: 'email', value: '12121$gmail.com'}});
      expect(wrapper.state('email')).toEqual('12121$gmail.com');
     })
   
     it('should respond to changed  event and change the state of the SignUp Component', () => {
      wrapper.find('#password').simulate('change', {target: {name: 'password', value: '-----'}});
      expect(wrapper.state('password')).toEqual('-----');
     })
 
})