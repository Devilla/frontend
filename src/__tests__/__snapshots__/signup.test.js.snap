// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>SignUp --- Snapshot +++capturing Snapshot of SignUp 1`] = `
<div>
  <div
    className="authpage section innerpage"
  >
    <div
      className="wrapper"
    >
      <div
        className="animated fadeIn leftwrap center"
        style={
          Object {
            "animationDelay": "0ms",
            "pointerEvents": "all",
          }
        }
      >
        <form
          className="loginfrm"
          onSubmit={[Function]}
        >
          <div
            className="main-container"
          >
            <section
              className="switchable bg--secondary"
            >
              <div
                className="container"
              >
                <div
                  className="row justify-content-between align-items-center"
                >
                  <div
                    className="col-md-6 col-lg-6 text-center"
                  >
                    <div
                      className="switchable__text"
                    >
                      <h3>
                        Your first step towards conversions, Signup here.
                      </h3>
                      <span
                        className=" type--fine-print"
                      >
                        Already have an account? 
                        <a
                          onClick={[Function]}
                          style={Object {}}
                        >
                          Sign in
                        </a>
                      </span>
                      
                      <hr
                        className="short"
                      />
                      <form>
                        <div
                          className="row"
                        >
                          <div
                            className="col-12"
                          >
                            <input
                              name="username"
                              onBlur={[Function]}
                              onChange={[Function]}
                              placeholder="Your Name"
                              type="text"
                            />
                            <span
                              className="help-block"
                            >
                              <p
                                className="website-error"
                              />
                            </span>
                          </div>
                          <div
                            className="col-12"
                          >
                            <input
                              name="email"
                              onBlur={[Function]}
                              onChange={[Function]}
                              placeholder="Email Address"
                              type="email"
                              value=""
                            />
                            <span
                              className="help-block"
                            >
                              <p
                                className="website-error"
                              >
                                
                              </p>
                            </span>
                          </div>
                          <div
                            className="col-12"
                          >
                            <input
                              maxLength={20}
                              name="password"
                              onBlur={[Function]}
                              onChange={[Function]}
                              placeholder="Password"
                              type="password"
                            />
                            <span
                              className="help-block"
                            >
                              <p
                                className="website-error"
                              >
                                
                              </p>
                            </span>
                          </div>
                          <div
                            className="col-12"
                          >
                            <input
                              maxLength={20}
                              name="confirmPassword"
                              onBlur={[Function]}
                              onChange={[Function]}
                              placeholder="Confirm Password"
                              type="password"
                            />
                            <span
                              className="help-block"
                            >
                              <p
                                className="website-error"
                              >
                                
                              </p>
                            </span>
                          </div>
                          <div
                            className="frmcntl col-12"
                          >
                            <input
                              className="button submit-button w-button btn btn--primary ml-0"
                              disabled={false}
                              type="submit"
                              value="Create Account"
                            />
                          </div>
                          <hr />
                          <div
                            className="col-12"
                          >
                            <span
                              className="type--fine-print"
                            >
                              By signing up, you agree to the 
                              <a
                                onClick={[Function]}
                                style={Object {}}
                              >
                                Terms of Service
                              </a>
                            </span>
                          </div>
                        </div>
                      </form>
                      <hr
                        className="short"
                      />
                      <p>
                         
                        <span
                          className=" type--fine-print b"
                        >
                          Or Signup using:  
                        </span>
                         
                      </p>
                      <div
                        className=""
                      >
                        <a
                          className="link "
                          href="undefinedconnect/facebook"
                        >
                          <a
                            className="btn btn--icon bg--facebook"
                            onClick={[Function]}
                            style={Object {}}
                          >
                            <span
                              className="btn__text "
                            >
                              <i
                                className="socicon socicon-facebook"
                              />
                              Signup using Facebook
                            </span>
                          </a>
                        </a>
                        <p />
                        <a
                          className="link"
                          href="undefinedconnect/google"
                        >
                          <a
                            className="btn btn--icon bg--googleplus"
                            onClick={[Function]}
                            style={Object {}}
                          >
                            <span
                              className="btn__text"
                            >
                              <i
                                className="socicon socicon-google"
                              />
                              Signup using Google
                            </span>
                          </a>
                        </a>
                      </div>
                    </div>
                  </div>
                  <div
                    className="col-md-8 col-lg-8 pr-0"
                  >
                    <img
                      alt="Image"
                      src="signuppage2.png"
                    />
                  </div>
                </div>
              </div>
            </section>
          </div>
        </form>
        <div
          className="support"
        />
      </div>
    </div>
  </div>
</div>
`;
